server_config.go

```diff
@@ -20,10 +20,10 @@ type RelayAddressGenerator interface {
 	Validate() error
 
 	// Allocate a PacketConn (UDP) RelayAddress
-	AllocatePacketConn(network string, requestedPort int) (net.PacketConn, net.Addr, error)
+	AllocatePacketConn(network string, requestedPort int, username string) (net.PacketConn, net.Addr, error)
```

Comment:
I'm afraid we will need to add the realm too. From RFC8856:
> Realm:
>    A string used to describe the server or a context within the server. The realm tells the client which username and password combination to use to authenticate requests.

Reply:
Thanks @rg0now for the comment. I thought about it too but because a pion/turn's turn.Server instance can only have one realm, once the allocate request is authorized by the AuthHandler (which would consume the realm from the client), the relay address allocation would not need it because at that point, USERNAME is sufficient to identify the requesting (and authorized) user IMO. I maybe missing something. Please let me know if you had any more thoughts.

Reply:
Fair enough. But then again, due to the same reason (one realm per server) we would not need the `realm` in the `authHandler` either. I think the idea is that for implementations running multiple server instances we can still have a single `authHandler` function instead of having to maintain a separate one for every server instance. But I agree that the case here is somewhat subtle.

Reply:
> we would not need the realm in the authHandler either.

It is up to the implementor of the authHandler but I would check if the realm matches, not only because TURN client has to reply with the same realm in the challenge (with 401) as required, but also the client might be misbehaving and sending the request to a wrong server.